src/agent/src/watcher.rs

```diff
@@ -0,0 +1,487 @@
+// Copyright (c) 2021 Apple Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::time::SystemTime;
+
+use tokio::fs;
+use tokio::sync::Mutex;
+use tokio::task;
+use tokio::time::{self, Duration};
+
+use anyhow::{ensure, Context, Result};
+use async_recursion::async_recursion;
+use nix::mount::{umount, MsFlags};
+use slog::{error, Logger};
+
+use crate::mount::BareMount;
+use crate::protocols::agent::Storage;
+
+/// The maximum number of file system entries agent will watch for each mount.
+const MAX_ENTRIES_PER_STORAGE: usize = 8;
```

Comment:
As mentioned, maybe this should be configurable with `8` being the default?

Reply:
I'm inclined to start opinionated with eight and make it configurable if/when we have a use case for it. This way we can keep it simple to start; I'm not sure we'll ever need to adjust this, and it'll avoid having to pass this information from runtime to agent, as well as the config scaffolding. WDYT? Am I just being lazy?